### PR TITLE
OCPBUGS-32495: prep stage doc updates and unify logs and naming

### PR DESCRIPTION
**Pull request by pixelsoccupied**
# Background / Context

This PR finalizes the work related Prep stage resiliency with docs + log messages + uniform naming convention  

Things to note: 
- Cleaned up existing docs to reflect the updates better and helpful commands to get logs.
- Removed precache plugin md file as they were mostly dev focused (all which are already in .go file) and/or outdated. 
- Streamlined logs and status messages to include job metadata 
- Updated precache resource name to be simply `lca-prep-precache`. This should indicate that the job is from `Prep` stage.  This also means previously appended resource names (`job` or `cm`) are also dropped as they are redundant and they are both using the same name. 
- ~~Updated extra-manifest warning annotation key to follow convention. `<subsystem>.lca.openshift.io/warn-<name of the warning>`~~
- ~~Addressed nits from prev PR~~
(Strikethrough will be addressed in another PR)

Ref: 
https://github.com/openshift-kni/lifecycle-agent/pull/476
https://github.com/openshift-kni/lifecycle-agent/pull/498

/cc @donpenney @Missxiaoguo

### CODE REVIEW

**Comment by pixelsoccupied** [comment]
/hold 
until https://github.com/openshift-kni/lifecycle-agent/pull/498

**Comment by pixelsoccupied** [comment]
/hold 
testing a little bit

File: internal/extramanifest/extramanifest.go [hunk 1]
```diff
@@ -92,13 +92,6 @@ func NewEMFailedValidationError(msg string) *EMStatusError {
 	}
 }
 
-func NewEMUnknownCRDError(msg string) *EMStatusError {
-	return &EMStatusError{
-		Reason:     "UnknownCRD",
```
Comment (pixelsoccupied) [inline internal/extramanifest/extramanifest.go:97]: Dropping these changes for another PR

**Comment by pixelsoccupied** [comment]
the extramanifest clean up/nit covered in https://github.com/openshift-kni/lifecycle-agent/pull/521. wdyt about getting this PR in before 521?

**Review by donpenney — APPROVED** [review]

**Pull request merged**